src/chains/ethereum/ethereum/src/helpers/run-call.ts

```diff
@@ -128,29 +121,23 @@ export async function applySimulationOverrides(
   vm: VM,
   overrides: CallOverrides
 ): Promise<void> {
-  const stateManager = vm.stateManager;
+  const eei = vm.eei;
   for (const address in overrides) {
     if (!hasOwn(overrides, address)) continue;
     const { balance, nonce, code, state, stateDiff } = overrides[address];
 
-    const vmAddr = { buf: Address.from(address).toBuffer() } as any;
+    const vmAddr = {
+      buf: Address.from(address).toBuffer()
+    } as EthereumJsAddress;
     // group together overrides that update the account
     if (nonce != null || balance != null || code != null) {
-      const account = await stateManager.getAccount(vmAddr);
+      const account = await eei.getAccount(vmAddr);
 
       if (nonce != null) {
-        account.nonce = {
-          toArrayLike: () =>
-            // geth treats empty strings as "0x0" nonce for overrides
-            nonce === "" ? BUFFER_EMPTY : Quantity.toBuffer(nonce)
-        } as any;
+        account.nonce = Quantity.toBigInt(nonce === "" ? "0x0" : nonce);
       }
       if (balance != null) {
-        account.balance = {
-          toArrayLike: () =>
-            // geth treats empty strings as "0x0" balance for overrides
-            balance === "" ? BUFFER_EMPTY : Quantity.toBuffer(balance)
-        } as any;
+        account.balance = Quantity.toBigInt(balance === "" ? "0x0" : balance);
```

Comment:
```suggestion
        account.balance = balance === "" ? 0n : Quantity.toBigInt(balance);
```
skips the string=>buffer=>bigint conversion of "0x0"

Reply:
good find!